service/filter.go

```diff
@@ -86,13 +86,24 @@ func ParseRequestForm() Filter {
 	}
 }
 
+func isZero(length string) bool {
+	if length == "" {
+		return true
+	}
+	i, err := strconv.Atoi(length)
+	if err != nil {
+		return false
+	}
+	return i == 0
```

Comment:
The `i` may be negative.
You can merge the two conditions with:
```go
return err != nil || i <= 0
```

Reply:
i == 0 should return true...